# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1461191375 25200
#      Wed Apr 20 15:29:35 2016 -0700
# Node ID a3a412dee70450260bf53d305b824a6d1700eb0c
# Parent  986dbd769bb1e785eefe46f3796625436b8919a3
* parser.tst: Add BIST tests for 64-bit integer literal support (bug #47690)

diff --git a/test/parser.tst b/test/parser.tst
--- a/test/parser.tst
+++ b/test/parser.tst
@@ -282,16 +282,22 @@
 %!assert (0xAB_CD, 43981)
 %!assert (2e0_1, 20)
 
 ## Test binary constants
 %!assert (0b101, 5)
 %!assert (0B1100_0001, 0xC1)
 %!assert (class (0b1), "double")
 
+## Test range of large binary and hexadecimal literals
+%!assert (0x8000_0000_0000_0000, 2^63)
+%!assert (0xFFFF_FFFF_FFFF_FFFF, 2^64)
+%!assert (0b10000000_0000000_000000000_00000000_00000000_00000000_00000000_00000000, 2^63)
+%!assert (0b11111111_1111111_111111111_11111111_11111111_11111111_11111111_11111111, 2^64)
+
 ## Test creation of anonymous functions
 
 %!test
 %! af_in_cell = {@(x) [1 2]};
 %! assert (af_in_cell{1}(), [1, 2]);
 
 %!test
 %! R = @(rot) [cos(rot) -sin(rot); sin(rot) cos(rot)];
